app/src/main/java/org/dhis2/utils/session/PinPresenter.kt

```diff
@@ -2,10 +2,14 @@ package org.dhis2.utils.session
 
 import org.dhis2.data.prefs.Preference
 import org.dhis2.data.prefs.PreferenceProvider
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.D2Manager
+import java.lang.Exception
 
 class PinPresenter(
     val view: PinView,
-    val preferenceProvider: PreferenceProvider
+    val preferenceProvider: PreferenceProvider,
+    val d2: D2? = D2Manager.getD2()
```

Comment:
for me it's okay to inject d2 doing it this way. But since we're using dagger to maintain consistency, could we use Pin module instead?

Reply:
The Pin module is outside the user session, so it is not possible to inject d2. I think we can refactor this once we review the full pin logic. What do you think?